Source/WebCore/css/parser/CSSParserContext.cpp

```diff
@@ -146,7 +146,7 @@ void add(Hasher& hasher, const CSSParserContext& context)
         | context.viewTransitionTypesEnabled                << 24
         | context.cssProgressFunctionEnabled                << 25
         | (uint32_t)context.mode                            << 26; // This is multiple bits, so keep it last.
-    add(hasher, context.baseURL, context.charset, context.propertySettings, bits);
+    add(hasher, context.baseURL, AtomString(context.charset), context.propertySettings, bits);
```

Comment:
@cdumez Thoughts on this? You use a bespoke hasher in TextEncodingRegistry, which maybe should be part of Hasher.h so it could just accept ASCIILiteral.

I though AtomString might be good here, since it would presumably hash to a consistent value and would amortize the cost of casting to string since these encoding are always one of a small set.

Reply:
We should be able to add this to Hasher.h or ASCIILiteral.h so that we can just pass an ASCIILiteral in:
```
inline void add(Hasher& hasher, ASCIILiteral literal)
{
    // Chose to hash the characters here. Assuming this is better than hashing the possibly-already-computed hash of the characters.
    bool remainder = literal.length() & 1;
    unsigned roundedLength = literal.length() - remainder;
    for (unsigned i = 0; i < roundedLength; i += 2)
        add(hasher, (literal[i] << 16) | literal[i + 1]);
    if (remainder)
        add(hasher, literal[roundedLength]);
}
```

Reply:
Great! I will add that.